digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -536,59 +536,49 @@ trait LaneOperations {
     * @param username Username of the user is doing the changes
     * @return
     */
-  def update ( newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long] ,sideCode: Int, username: String ): Seq[Long] = {
-    if (newIncomeLane.isEmpty || linkIds.isEmpty)
-      return Seq()
+  def update(newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long], sideCode: Int, username: String, newTransaction: Boolean = true): Seq[Long] = {
+    def updateProcess(): Seq[Long] = {
+      if (newIncomeLane.isEmpty || linkIds.isEmpty)
```

Comment:
Maybe we can do this validation before if (newTransaction)

Reply:
Done